corehq/apps/linked_domain/tests/test_update_custom_data_fields.py

```diff
@@ -85,6 +90,189 @@ def test_update(self):
         model = CustomDataFieldsDefinition.objects.get(domain=self.linked_domain,
                                                        field_type=UserFieldsView.field_type)
         profiles_by_name = {p.name: p for p in model.get_profiles()}
-        self.assertTrue(bool(profiles_by_name[self.coral_profile.name]))    # updating doesn't delete profiles
         self.assertEqual(profiles_by_name[self.fish_profile.name].fields, self.fish_profile.fields)
         self.assertEqual(profiles_by_name[lamprey_profile.name].fields, lamprey_profile.fields)
+
+
+class TestUpdateFields(TestCase):
+    def test_can_update_user_fields(self):
+        self._set_fields([Field(slug='a', label='label1', is_synced=True)], UserFieldsView.field_type)
+
+        updated_field = Field(slug='a', label='label2')
+        update_definition = self._generate_update_definition_for_fields([updated_field], UserFieldsView.field_type)
+        update_custom_data_models_impl(update_definition, self.domain)
+
+        fields = self._get_fields(UserFieldsView.field_type)
+        expected_field = Field(slug='a', label='label2', is_synced=True)
+        self.assertEqual(len(fields), 1)
+        self.assertEqual(fields[0].to_dict(), expected_field.to_dict())
```

Comment:
nit: I often find failure output to be more informative when it shows more than a simple length mismatch. Consider this alternative:
```suggestion
        self.assertEqual([f.to_dict() for f in fields], [expected_field.to_dict()])
```
The same applies to other tests as well.

Reply:
It may be just how I interpret things, but I think reframing it this way obscures the assertion. For me, the two assertions state: check that there is exactly 1 field, and that it matches what is specified. When I see the list comprehension, despite knowing what the test is supposed to do, I think I start to question why we're iterating over all fields when I only want to look at one. 

I think the other thing is that I don't necessarily expect failing tests to give the data necessary to correct the test; rather, they should provide enough information to understand what is going wrong and start investigating. I think seeing that the test expected 1 record, but received 2, is straight-forward enough to begin investigating.